internal/services/datafactory/data_factory.go

```diff
@@ -520,79 +520,36 @@ func flattenDataFactoryDatasetSFTPLocation(input *datafactory.SftpLocation) []in
 	return []interface{}{result}
 }
 
-func flattenDataFactoryDatasetCompression(input datafactory.BasicDatasetCompression) []interface{} {
+func flattenDataFactoryDatasetCompression(input *datafactory.DatasetCompression) []interface{} {
 	if input == nil {
 		return nil
 	}
 	result := make(map[string]interface{})
 
-	if compression, ok := input.AsDatasetBZip2Compression(); ok {
-		result["type"] = compression.Type
-	}
-	if compression, ok := input.AsDatasetDeflateCompression(); ok {
-		result["type"] = compression.Type
-	}
-	if compression, ok := input.AsDatasetGZipCompression(); ok {
-		result["type"] = compression.Type
-		result["level"] = compression.Level
-	}
-	if compression, ok := input.AsDatasetTarCompression(); ok {
-		result["type"] = compression.Type
-	}
-	if compression, ok := input.AsDatasetTarGZipCompression(); ok {
-		result["type"] = compression.Type
-		result["level"] = compression.Level
-	}
-	if compression, ok := input.AsDatasetZipDeflateCompression(); ok {
-		result["type"] = compression.Type
-		result["level"] = compression.Level
+	result["type"] = input.Type
+	var level string
+
+	if input.Level != nil {
+		level = input.Level.(string)
```

Comment:
this'd also work:

```suggestion
		level = string(input.Level)
```

Reply:
I'm afraid String() is not gonna work since the input value is an interface type.